src/rtp_table.h

```diff
@@ -1,43 +0,0 @@
-/*
- * This file is part of EasyRPG Player.
- *
- * EasyRPG Player is free software: you can redistribute it and/or modify
- * it under the terms of the GNU General Public License as published by
- * the Free Software Foundation, either version 3 of the License, or
- * (at your option) any later version.
- *
- * EasyRPG Player is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with EasyRPG Player. If not, see <http://www.gnu.org/licenses/>.
- */
-
-#ifndef EP_RTP_TABLE_H
-#define EP_RTP_TABLE_H
-
-#include <map>
-#include <string>
-#include <cstring>
-
-namespace RTP {
-	struct rtp_table_cmp {
-		bool operator()(char const* a, char const* b) const {
-			return std::strcmp(a, b) < 0;
-		}
-	};
-
-	using rtp_table_type = std::map<const char*, std::map<const char*, const char*, rtp_table_cmp>, rtp_table_cmp>;
```

Comment:
Since you're constructing this all at startup, you'd get better performance using `std::lower_bound` on sorted vectors than you would with `std::map`.

The problem with `map` is that it's a node based container, which means a separate heap allocation for each node. This uses more memory and even worse has terribly cache locality.

To pull this off cleanly, you' have to make your `rtp_table_type` an actual class with these semantics embedded inside. Boost provides something called a `flat_map` which does this, but we don't want to use boost and implementing a simplified version of this for rtp table is trivial.

If there is lack of time for 6.1, we could punt this optimization for the next PR.

Reply:
The std::map RTP code is deleted as part of this PR. The new data structure is 2D-char*-array / table.